processor/resourcedetectionprocessor/internal/resourcedetection.go

```diff
@@ -194,6 +208,21 @@ func MergeSchemaURL(currentSchemaURL string, newSchemaURL string) string {
 	return currentSchemaURL
 }
 
+func FilterAttributes(am pdata.AttributeMap, attributesToKeep map[string]struct{}) []string {
```

Comment:
Does this function need to be exported?

Reply:
I have no strong feelings one way or the other